src/core/components/InstallButton/index.js

```diff
@@ -71,6 +70,18 @@ export class InstallButtonBase extends React.Component {
     return i18n.sprintf(label, { name });
   }
 
+  getDownloadProgress() {
```

Comment:
I have a couple thoughts on this:

1. This function / logic is only used by `Switch`. Why not move it there?

2. I think it's a bad pattern to pass multiple properties through a component simply because a child component needs them, i.e. passing `status` and `downloadProgress` from `Addon` to `InstallButton` to `Switch`. Using redux to fetch download progress from the state only where it's needed (i.e. `Switch`) would solve this problem nicely. The way I would do it would be to create a `downloadTransactionId` in the `Addon` parent when the user initiates installation then pass that property all the way down: `InstallButton` > `Switch`. The `Addon` could continue dispatching download progress data but it would need to add a transaction ID. `Switch` could fetch download progress directly from the state based on `downloadTransactionId`. This model would still keep `Switch` relatively dumb because it wouldn't know anything about add-ons or versions, only about the current download transaction. The key difference in this model is that the entire chain of components is only re-rendered *once* when the `downloadTransactionId` changes, not all the time as `downloadProgress` gets updated repeatedly.

My first suggestion is probably worth doing (if you agree). The second suggestion is a much bigger change and probably not worth spending time on today.